src/components/FormSelectList.vue

```diff
@@ -187,7 +187,11 @@ export default {
       const { selectedEndPoint, selectedDataSource, dataName } = options;
 
       // If no data source has been specified, do not make the api call
-      if (!!selectedDataSource && selectedDataSource.toString().trim().length === 0) {
+      if (
+        selectedDataSource === null ||
+        typeof selectedDataSource === "undefined" ||
```

Comment:
```suggestion
        !!selectedDataSource ||
```

Reply:
truth tables are not the same:

| selectedDataSource | selectedDataSource === null \|\| typeof selectedDataSource === "undefined" | !!selectedDataSource |
| ------------------ | ------------------------------------------------------------------------ | -------------------- |
| null               | true                                                                     | false                |
| undefined          | true                                                                     | false                |
| object             | false                                                                    | true                 |